### ISSUE TITLE
Windows Defender blocks Bleachbit

### ISSUE THREAD

**Issue by cliftonprince** [bleachbit/bleachbit#1004]
New error, can't remove Windows Defender backups / updates (formerly could do so, now gives following errors):

[WinError 5] Access is denied.: Command to delete C:\ProgramData\Microsoft\Windows Defender\Definition Updates\Backup\mpasbase.vdm
[WinError 5] Access is denied.: Command to delete C:\ProgramData\Microsoft\Windows Defender\Definition Updates\Backup\mpasdlta.vdm
[WinError 5] Access is denied.: Command to delete C:\ProgramData\Microsoft\Windows Defender\Definition Updates\Backup\mpavbase.vdm
[WinError 5] Access is denied.: Command to delete C:\ProgramData\Microsoft\Windows Defender\Definition Updates\Backup\mpavdlta.vdm
[WinError 5] Access is denied.: Command to delete C:\ProgramData\Microsoft\Windows Defender\Definition Updates\Backup\mpengine.dll

**Comment by az0** [comment]
Which version of BleachBit is this? Maybe 4.0.0 final, 4.1.0 beta, or another version?

**Comment by the-cleaner-66** [comment]
Same errors here, test with BB 4.0.0 final and BB 4.1.0.1723 beta (both portable) on Windows 10 1809, build 17763_rs5

**Comment by az0** [comment]
@the-cleaner-66 
Is it identical with both `access is denied` and the `C:\ProgramData\Microsoft\Windows Defender\Definition Updates\Backup` part? Does this happen only when the "Backup" option is enabled under the "Windows Defender" section?

**Comment by the-cleaner-66** [comment]
@az0 
Yes, in fact, it's identical with both parts. The "Backup" option was all the time enabled. The error happened only in the last 6...8 weeks.

**Comment by the-cleaner-66** [comment]
@az0 
PS: No error appears if Win 10 is booting into safe mode...

**Comment by az0** [comment]
I presume that an active process has the file locked, so BleachBit cannot delete it. Normally, BleachBit would fall back to queuing it for deleting after a system reboot, but it seems this isn't working here.

**Comment by az0** [comment]
I am closing this issue because Windows Defender "defends" its files, preventing tampering, so not much can be done, but you can delete them in safe mode. This is better than, for example, not offering to delete them, ever.

Important: make sure to enable safe mode _with networking_ or BleachBit [may not start](https://github.com/bleachbit/bleachbit/issues/1785).

By the way, this week I committed a change to make the error message prettier. They were multiple lines, which seems to be a regression, but now they are shorter---not even the "WinError".
